python/ql/src/Security/CWE-295/RequestWithoutValidation.ql

```diff
@@ -0,0 +1,36 @@
+/**
+ * @name Request without certificate validation
+ * @description Making a request without certificate validation can allow man-in-the-middle attacks.
+ * @kind problem
+ * @problem.severity error
+ * @precision medium
+ * @id py/request-without-cert-validation
+ * @tags security
+ *       external/cwe/cwe-295
+ */
+
+import python
+
+import semmle.python.web.Http
+
+
+FunctionObject requestFunction() {
+    exists(ModuleObject req |
+        req.getName() = "requests" and
+        result = req.getAttribute(httpVerbLower())
+    )
```

Comment:
This could be written a bit shorter as `result = any(ModuleObject req | req.getName() = "requests").getAttribute(httpVerbLower())`.

We should consider extending the library to have an easier way of expressing the `any(ModuleObject m | m.getName() = "foo")` idiom, as it is quite common. Perhaps we could add a `NamedModule` subclass of `ModuleObject` that accepts a string `name` in its characteristic predicate, and adds `this.getName() = name` as a constraint?

Reply:
See https://github.com/Semmle/ql/pull/554